raspberry/checkout/checkout_main.py

```diff
@@ -34,6 +36,7 @@ def server_response_received(self, response):
             if response == STATUS_NOK:
                 print("Checkout failed")
                 self.interactions.indicate_error()
+                self.logic.remove_last_scanned()
```

Comment:
if anything, here should be cancelling the chekcout or no action. This response is for sending the summary to the server and buying items, not for getting a fedback after scanning the single tag.